mbzirc_ign/src/RFRange.cc

```diff
@@ -0,0 +1,534 @@
+    std::normal_distribution<> d{rxPowerDist.mean, sqrt(rxPowerDist.variance)};
+    rxPower = d(this->rndEngine);
+  }
+
+  // Based on rx_power, and noise value, compute the bit error rate (BER).
+  double ber = this->QPSKPowerToBER(
+    this->DbmToPow(rxPower), this->DbmToPow(this->radioConfig.noiseFloor));
+
+  double packetDropProb = 1.0 - exp(_numBytes * log(1 - ber));
+
+  // igndbg << "TX power (dBm): " << this->radioConfig.txPower << "\n" <<
+  //           "RX power (dBm): " << rxPower << "\n" <<
+  //           "BER: " << ber << "\n" <<
+  //           "# Bytes: " << _numBytes << "\n" <<
+  //           "PER: " << packetDropProb << std::endl;
+
+  double randDraw = ignition::math::Rand::DblUniform();
+  bool packetReceived = randDraw > packetDropProb;
+
+  if (!packetReceived)
+    return std::make_tuple(false, std::numeric_limits<double>::lowest());
+
+  return std::make_tuple(true, rxPower);
+}
+
+//////////////////////////////////////////////////
+void RFRangePrivate::UpdateSensorPoses(gazebo::EntityComponentManager &_ecm)
+{
+  for (auto &[entityId, data] : this->entityMap)
+  {
+    // Pose of the model.
+    auto modelPose = _ecm.Component<components::Pose>(entityId);
+    if (modelPose == nullptr)
+      _ecm.CreateComponent(entityId, components::Pose());
+
+    // We just created the joint position component, give one iteration for the
+    // physics system to update its size
+    if (modelPose == nullptr)
+      continue;
+
+    data.pose = modelPose->Data();
+  }
+}
+
+//////////////////////////////////////////////////
+std::string RFRangePrivate::RemoveAllScope(const std::string &_name,
+  const std::string &_delim)
+{
+  auto sepPos = _name.rfind(_delim);
+  if (sepPos == std::string::npos || (sepPos + _delim.size()) > _name.size())
+    return _name;
+
+  return _name.substr(sepPos + _delim.size());
+}
+
+//////////////////////////////////////////////////
+RFRange::RFRange()
+  : dataPtr(std::make_unique<RFRangePrivate>())
+{
+}
+
+//////////////////////////////////////////////////
+void RFRange::Configure(const Entity &_entity,
+    const std::shared_ptr<const sdf::Element> &_sdf,
+    EntityComponentManager &_ecm,
+    EventManager &/*_eventMgr*/)
+{
+  this->dataPtr->Load(_sdf);
+
+  // Initialize system update period
+  double rate = _sdf->Get<double>("update_rate", 1).first;
+  std::chrono::duration<double> period{rate > 0 ? 1 / rate : 0};
+  this->dataPtr->updatePeriod =
+      std::chrono::duration_cast<std::chrono::steady_clock::duration>(period);
+}
+
+//////////////////////////////////////////////////
+void RFRange::PreUpdate(
+    const ignition::gazebo::UpdateInfo &_info,
+    ignition::gazebo::EntityComponentManager &_ecm)
+{
+  IGN_PROFILE("RFRange::PreUpdate");
+
+  _ecm.EachNew<RFRangeType,
+               ignition::gazebo::components::WorldPose>(
+    [&](const ignition::gazebo::Entity &_entity,
+        const RFRangeType *_custom,
+        const ignition::gazebo::components::WorldPose *_pose)->bool
```

Comment:
`_pose` not used?

Reply:
Removed in [8450745](https://github.com/osrf/mbzirc/pull/151/commits/8450745fb3e3afdf3bc481d5f44c0bf25c237b37).